### DISCUSSION TITLE
Change affine in StatefulTractogram

### DISCUSSION THREAD

**Discussion by HilaGast** [dipy/dipy#2412] [Tractography / Tractometry]
Hi, 
I have a trk file that saved from ExploreDTI and it seems like the file was not save correctly. While loading, it raises an error that tng process it automatically load streamlines with identity affine matrix.
I do have nifti file in the appropriate space. 
Is there any way to modify StatfulTractogram so I'll be able to save it with a proper header? Or anyone familiar with a way to fix this problem? 

Thanks, 
Hila

**Comment by skoudoro** [comment]
Hi @HilaGast,

Just in case you did not solve your issue, I supposed you can create a new StatefulTractogram with the old one

```new_sft = StatefulTractogram(old_sft.streamlines, reference=your_nifti_files)```

the header should be correct for new_sft.

I hope it helps!

ps: I convert this issue to a discussion
